[C팀] 프론트엔드파트 코드리뷰용 PR

yeyounging opened the pull request:
안녕하세요. 멋진 프론트엔드가 되기 위해 열심히 공부중인 프론트엔드 공예영, 황동준입니다.
따끔하고 따듯한 충언을 주신다면 뼈에 새기도록 하겠습니다.

귀한 시간 내어주셔서 감사합니다.

yeyounging committed:
setting: eslint 및 storybook 환경설정

nebulaBdj committed:
init: 프리티어 모든 파일에 적용, 도커 이미지 빌드, Next.js 컨테이너화

nebulaBdj committed:
init: GitHub Action CI/CD 구현 코드 작성

nebulaBdj committed:
init: deploy.yml 최종 코드

yeyounging committed:
setting: 폰트 및 tailwind 설정 추가

nebulaBdj committed:
init: CI/CD 구축 코드 수정

누락된 docker pull하는 코드 추가

nebulaBdj committed:
Merge branch 'main' of https://github.com/KUSITMS-30th-TEAM-C/frontend

nebulaBdj committed:
init: CI/CD 코드 수정

nebulaBdj committed:
init: CI/CD 코드 수정

yeyounging committed:
setting: framer-motion, zustand

yeyounging committed:
feat: 온보딩 페이지

yeyounging committed:
fix: public 이미지 경로 이슈

yeyounging committed:
setting: PR 빌드 자동화

yeyounging committed:
fix: 이미지 경로 이슈

yeyounging committed:
fix

yeyounging committed:
fix: public 폴더 배포를 위한 Dockerfile 업데이트

yeyounging committed:
Revert "fix: public 폴더 배포를 위한 Dockerfile 업데이트"

This reverts commit 951bf9363b29936c4036b3508957d0a8dca10630.

nebulaBdj committed:
Update deploy.yml

nebulaBdj committed:
Merge pull request #2 from KUSITMS-30th-TEAM-C/feat/#1_온보딩프로세스구현

feat : 온보딩프로세스구현

nebulaBdj committed:
Update Dockerfile

yeyounging committed:
Update Dockerfile

nebulaBdj committed:
Update Dockerfile

yeyounging committed:
Update Dockerfile

nebulaBdj committed:
init: CI/CD 코드 수정

yeyounging committed:
HOTFIX

yeyounging committed:
HOTFIX

nebulaBdj committed:
Create pull_request_template.md

nebulaBdj committed:
Update pull_request_template.md

nebulaBdj committed:
Update pull_request_template.md

nebulaBdj committed:
Init: 이슈 템플릿 생성

nebulaBdj committed:
feat: (#3) 스플래시 페이지 및 소셜 로그인 구현 (#5)

* feat: 스플래시 페이지 구현

스플래시 화면 구현, 해당 과정에서 폰트적용이 안돼 폰트 적용 부분도 수정함

* fix: 빌드 에러 해결

1. 불필요한 import 정리
2. button type 속성 추가

* feat: 소셜로그인 구현 완료

로컬 환경에서 테스트 완료

* fix: ESLint 규칙에 맞춰서 수정

추가적으로 로그인 페이지 SplashTop 위치도 수정

* fix: 빌드 에러 수정

* fix: useSearchParams가 서버에서 동작하지 않도록 Suspense로 감쌈

* fix: 요청 API 환경변수 설정, 불필요한 객체의 인스턴스 제거

yeyounging committed:
feat: 홈화면 디폴트 구현 (#12)

yeyounging committed:
feat: axios, 데이터페칭 환경설정 및 빠른시작 api 연결 (#14)

* setting: tanstack-query, AsyncBoundary를 이용한 데이터 페칭 환경설정

* setting: axios 환경설정

* feat: 빠른시작 api 연결

* feat : 빠른시작 등록 api

* fix: 테스트를 위한 마스터토큰적용

* fix: build 오류 수정

hyesungoh commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,24 @@
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: 레포지토리 체크아웃 중..
+        uses: actions/checkout@v2
+
+      - name: node 설치중..
+        uses: actions/setup-node@v2
+        with:
+          node-version: '18.18.2'
```
각 액션들 버전을 올리면 좋을 거 같아요 ~ 
```suggestion
    steps:
      - name: 레포지토리 체크아웃 중..
        uses: actions/checkout@v4

      - name: node 설치중..
        uses: actions/setup-node@v4
        with:
          node-version: '18.18.2'
```

hyesungoh commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,24 @@
+        with:
+          node-version: '18.18.2'
+
+      - name: 종속성 설치중...
+        run: npm install
+     
+      - name: 빌드 중..
+        run: npm run build
```
npm을 사용하시는 이유가 궁금해용

hyesungoh commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,62 @@
+    name: Push to mcp container registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout # GitHub Action은 해당 프로젝트가 만들어진 환경에서 checkout하고 나서 실행
+        uses: actions/checkout@v3
```
여기는 체크아웃 버전 3이네요 ㅋㅋㅋ

hyesungoh commented on Dockerfile:
```diff
@@ -0,0 +1,85 @@
+# 프리티어 오류 발생으로 빌드 전에 프리티어 규칙 적용
+RUN npx prettier --write .
+
+RUN npm run build
+
+
+
```
도커에 업로드하고 빌드하는 것보다, cd 단계에서 빌드하고 빌드된 결과물만 도커에 올리는 것이 훨씬 빨라용

hyesungoh commented on package.json:
```diff
@@ -6,31 +6,54 @@
   },
   "dependencies": {
+    "@tanstack/react-query": "^5.59.18",
+    "@types/js-cookie": "^3.0.6",
```
크게 중요하진 않지만, 이건 개발 의존성인 거 같아요 ~

hyesungoh commented on src/react-utils/AsyncBoundary/index.tsx:
토스의 경우 AsyncBoundary를 걷어내고 있는 추세라고 하더라고요

관리주체가 토스가 된 suspensive도 v2에서 AsyncBoundary를 제거했고요

관련한 이유는 서스펜시브 공식 문서를 확인해보시면 좋을 거 같아요 ~~

> 그렇다고 이런 형태가 잘못되었다는 것은 아닙니다 👍

hyesungoh commented on src/components/common/Badge/index.tsx:
```diff
@@ -0,0 +1,19 @@
+import { StrictPropsWithChildren } from '@/types'
+import { cn } from '@/util'
+
+interface IBadge extends StrictPropsWithChildren {
+  className?: string
+}
```
헝가리안 표기법을 사용하시는 이유가 있을까요?

> 구글 타입스크립트 스타일 가이드에서는 지양하도록 기술되어 있어요 https://google.github.io/styleguide/tsguide.html#descriptive-names

hyesungoh commented on src/types/react.ts:
```diff
@@ -1,13 +1,7 @@
+export type ReactRef<T> = RefObject<T> | MutableRefObject<T> | Ref<T>
 
-export type StrictPropsWithChildren<P = unknown> = P & { children: ReactNode };
+export type StrictPropsWithChildren<P = unknown> = P & { children: ReactNode }
```
이 타입이 필요한 이유를 모르겠어요

react의 `PropsWithChildren` 으로 대체할 수 있으며, 직접 ReactNode 타입을 기입하는게 더 직관적일 수 있을 거 같은데

사용하신 이유가 궁금해요

hyesungoh commented on src/components/common/Button/useButton.ts:
```diff
@@ -0,0 +1,46 @@
+
+  const handleClick = useCallback(
+    (e: MouseEvent<HTMLButtonElement>) => {
+      if (!disabled || !isLoading) {
+        onClick?.(e)
+      }
```
disabled 됐을 때 onClick이 호출되지 않는 것은 내장되어 있지 않나용?

hyesungoh commented on src/app/start/components/Step1.tsx:
step 같은 퍼널 형태는 일반적으로 많이 쓰여서,

적절한 추상화를 제공하는 도구들이 있어요

참고해보시면 좋을 거 같아요 ~ 

https://github.com/toss/use-funnel

hyesungoh commented on src/app/page.tsx:
```diff
@@ -1,5 +1,103 @@
+  const [logoColor, setlogoColor] = useState('white')
+  const [splashBoxColor, setSplashBoxColor] = useState('#31313B')
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setIsSplash(false)
+      setSplashBoxColor('#F3F3F4')
+    }, 1000)
+
+    const logotimer = setTimeout(() => {
+      setlogoColor('#1A1A25')
+    }, 1700)
+
+    return () => {
+      clearTimeout(timer)
+      clearTimeout(logotimer)
+    }
+  }, [])
```
이렇게 스플래시를 보여준다면 사용자 경험이 저해되지 않을까요? ㅋㅋㅋ

hyesungoh commented on src/app/page.tsx:
```diff
@@ -1,5 +1,103 @@
   return (
-    <div className="text-[#444444] flex justify-center items-center h-screen">hi~</div>
+    <div className=" flex justify-center items-center w-screen h-screen">
+      <motion.div
+        className=" relative w-full h-full bg-primary_foundation_100"
+        initial={{ opacity: 1 }}
+        animate={{
+          opacity: 1,
+          backgroundColor: isSplash ? '#1A1A25' : '#ffffff',
+        }}
+        transition={{ duration: 0.5 }}
+      >
```
lazy motion을 활용해보셔도 좋을 거 같아요 ~

https://www.framer.com/motion/lazy-motion/

yeyounging committed:
[feat] 홈 화면 / 빠른시작 ui 및 api 연결 (#17)

* feat: 홈화면 ui 및 데이터 페칭

* feat: 홈화면 분기별 ui

* feat: 빠른시작 등록, 리스트

* refactor

* fix: 라우팅 경로 오류 수정

yeyounging committed:
[refactor] axios 에러 응답 수정 및 온보딩 api 연결 로직 리팩토링 (#19)

* feat: 홈화면 ui 및 데이터 페칭

* feat: 홈화면 분기별 ui

* feat: 빠른시작 등록, 리스트

* refactor

* fix: 라우팅 경로 오류 수정

* feat: 온보딩 api 연결

* refactor: axios 에러 응답 반환 수정

* refactor: 에러처리

* fix: eslint

yeyounging committed:
[Feat] 아카이빙 캘린더, 컴포넌트 깜빡임 대응 (#21)

* feat: 홈화면 ui 및 데이터 페칭

* feat: 홈화면 분기별 ui

* feat: 빠른시작 등록, 리스트

* refactor

* fix: 라우팅 경로 오류 수정

* feat: 온보딩 api 연결

* refactor: axios 에러 응답 반환 수정

* refactor: 에러처리

* fix: eslint

* feat: Calendar컴포넌트, useCalendar훅 구현

* chore

* chore: 아이콘이름 변경

* chore: 오타 수정

* fix: 폰트 적용

* refactor: 기존 HomeHeader 재사용성을 위한 리팩토링

* feat: 아카이빙 레이아웃 로직 및 달력 컴포넌트, 활동 컴포넌트 구현

* feat: api 연결 및 깜빡임 이슈 대응

yeyounging committed:
feat: 닉네임 중복 검사 api 연결 (#24)

yeyounging committed:
[feat] 온보딩페이지 기획 변경사항 반영 (#25)

* feat: 닉네임 중복 검사 api 연결

* fix: 액세스토큰 쿠키에 저장

* feat: 기획안 반영

* fix: eslint

nebulaBdj committed:
[feat] (#11) 활동 추천 기능 구현 (#18)

* fix: 컴포넌트 폴더 라우팅 안되는 폴더로 변경

* feat: 활동 추천 기능 구현 중
자투리 시간 입력, 온/오프라인 선택 기능 구현 완
현재 위치 선택 기능, 키워드 선택 기능 구현 중

* fix: CI 오류 해결

* fix: CI 오류 수정

* docs: 폴더 구조 수정

* feat: 활동 추천 기능 타이머 제외 완성

- 자투리 시간 입력 완
- 온오프라인 선택 완
- 현재 위치 탐색 및 검색 완
- 키워드 선택 완
- 추천 받은 활동 캐러셀 구현 완
- 이 활동하기 버튼 시 새 탭이 생성되며, 활동 타이머 패이지로 넘어감
- 백엔드 API 구현 완료 시 연동 예정
- 키워드 이미지 svgr 적용 후 변경 예정
- 타이머 오류로 인해 수정 중

* feat: 활동 선택 후 타이머 구현 완료

- API 연동하면서 수정해야 함
- 이후 지금까지 브랜치랑 merge하여 흐름 수정할 예정
    - 위의 과정에서 타이머 리팩토링도 함께 진행할 예정 급하게 만드느라 조잡함

* fix: CI 오류 수정 중

* fix: CI 오류 수정

잘못된 클래스 이름 및 설정 수정

* fix: CI 오류 수정 중

* fix: CI 오류 수정 중

* fix: eslint 오류 수정

* fix: ESlint 오류 최종 수정

빌드 성공,
ESLint의 consistent-return : 함수 내에서 모든 코드 경로가 값을 반환하거나 반환하지 않아야 함을 요구합니다. 
-> useEffect의 초기값에도 클린업 함수를 넣어 조건 충족

yeyounging committed:
[feat] 홈화면 변경된 기획안 적용 (#27)

* fix: axios 로직 수정

* style: 폰트 추가

* style: 변경된 디자인 반영

* feat: 빠른시작 라우팅을 위한 로컬스토리지 저장

* chore: 이미지 url 변경

* feat: 빠른시작->액티비티 연결

* fix

yeyounging committed:
[feat] 아카이빙 캘린더 세부기능 구현 (#30)

* feat: 모달 구현, 달력 선택 구현

* feat: 캘린더 가입일 기준 렌더링을 위한 userInfo 속성 추가

* fix: eslint

* chore: 변수명 변경

nebulaBdj committed:
[feat] 활동 추천 API 연동 (#31)

* feat: 활동 추천 API 연동

[구현한 내용]
- 활동 추천 기능 전체 흐름 관련 API 연동 완료
- 시작 스플래시 페이지 변경된 UI 일부 적용

[구현해야 할 작업]
- 실제 로그인 토큰 기반으로 API 연동
- API 호출 방식 변경하면서 적용될 예정
- 활동 중 시작 페이지로 왔을 때, 활동 페이지로 넘어가는 것

* fix: 소셜 로그인 url 수정

* HOTFIX: CI/CD 환경 변수 추가

* fix: 노출된 API url 환경변수 처리

nebulaBdj committed:
HOFIX: CI/CD 코드 수정 (공백 제거)

nebulaBdj committed:
HOTFIX: CI/CD 코드 수정 (누락된 환경변수 추가)

nebulaBdj committed:
HOTFIX: 환경 변수 전달을 위한 Dockerfile 수정

nebulaBdj committed:
HOFIX: CI/CD 코드 수정 (환경 변수 적용 시점 변경)

nebulaBdj committed:
HOTFIX: 환경 변수 출력 코드 추가

nebulaBdj committed:
HOTFIX: url 직접 추가하여 작업 test

nebulaBdj committed:
HOTFIX: CI/CD 코드 수정

nebulaBdj committed:
HOTFIX: 환경 변수 전달을 위한 Dockerfile 수정

ENV PORT 3000
=> ENV PORT=3000

nebulaBdj committed:
HOFIX: CI/CD 코드 수정

=로 변경했던 내용 :로 수정
docker 빌드할때도 환경 변수 삽입

nebulaBdj committed:
HOTFIX: 환경 변수 적용 후 소셜 로그인에 환경 변수 사용

nebulaBdj committed:
HOTFIX: Dockerfile 수정 (빌드 캐시 제거 코드 추가)

nebulaBdj committed:
HOTFIX: CI/CD 코드 수정

:를 =로 변경

yeyounging committed:
HOTFIX

nebulaBdj committed:
HOTFIX: CI/CD 코드 uses 버전 통일

yeyounging committed:
HOTFIX

yeyounging committed:
Merge branch 'main' of https://github.com/KUSITMS-30th-TEAM-C/frontend

yeyounging committed:
HOTFIX

yeyounging committed:
Update deploy.yml

yeyounging committed:
HOTFIX

yeyounging committed:
Merge branch 'main' of https://github.com/KUSITMS-30th-TEAM-C/frontend
